includes/class-amp-story-post-type.php

```diff
@@ -341,7 +343,280 @@ public static function remove_classic_editor_link( $actions, $post ) {
+		$allowed_attr[] = 'object-position';
+		$allowed_attr[] = 'position';
+		$allowed_attr[] = 'top';
+		$allowed_attr[] = 'left';
+		$allowed_attr[] = 'transform';
+
+		/*
+		 * CSS attributes that accept URL data types.
+		 *
+		 * This is in accordance to the CSS spec and unrelated to
+		 * the sub-set of supported attributes above.
+		 *
+		 * See: https://developer.mozilla.org/en-US/docs/Web/CSS/url
+		 */
+		$css_url_data_types = [
+			'background',
+			'background-image',
+
+			'cursor',
+
+			'list-style',
+			'list-style-image',
+		];
+
+		if ( empty( $allowed_attr ) ) {
+			return $css;
+		}
+
+		$css = '';
+		foreach ( $css_array as $css_item ) {
+			if ( '' === $css_item ) {
+				continue;
+			}
+
+			$css_item        = trim( $css_item );
+			$css_test_string = $css_item;
+			$found           = false;
+			$url_attr        = false;
+
+			if ( strpos( $css_item, ':' ) === false ) {
+				$found = true;
+			} else {
+				$parts        = explode( ':', $css_item, 2 );
+				$css_selector = trim( $parts[0] );
+
+				if ( in_array( $css_selector, $allowed_attr, true ) ) {
+					$found    = true;
+					$url_attr = in_array( $css_selector, $css_url_data_types, true );
+				}
+			}
+
+			if ( $found && $url_attr ) {
+				// Simplified: matches the sequence `url(*)`.
+				preg_match_all( '/url\([^)]+\)/', $parts[1], $url_matches );
+
+				foreach ( $url_matches[0] as $url_match ) {
+					// Clean up the URL from each of the matches above.
+					preg_match( '/^url\(\s*([\'\"]?)(.*)(\g1)\s*\)$/', $url_match, $url_pieces );
+
+					if ( empty( $url_pieces[2] ) ) {
+						$found = false;
+						break;
+					}
+
+					$url = trim( $url_pieces[2] );
+
+					if ( empty( $url ) || wp_kses_bad_protocol( $url, $allowed_protocols ) !== $url ) {
+						$found = false;
+						break;
+					} else {
+						// Remove the whole `url(*)` bit that was matched above from the CSS.
+						$css_test_string = str_replace( $url_match, '', $css_test_string );
+					}
+				}
+			}
+
+			if ( $found ) {
+				if ( '' !== $css ) {
+					$css .= ';';
+				}
+
+				$css .= $css_item;
+			}
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Filters the response before executing any REST API callbacks.
+	 *
+	 * Temporarily modifies post content during saving in a way that KSES
+	 * does not strip actually valid CSS from post content, making block content invalid.
+	 *
+	 * @todo Remove once core has better CSS parsing.
+	 *
+	 * @link https://core.trac.wordpress.org/ticket/37134
+	 *
+	 * @param WP_HTTP_Response|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
+	 * @param array                     $handler  Route handler used for the request.
+	 * @param WP_REST_Request           $request  Request used to generate the response.
+	 *
+	 * @return WP_HTTP_Response|WP_Error The filtered response.
+	 */
+	public static function filter_rest_request_for_kses( $response, $handler, $request ) {
+		if ( is_wp_error( $response ) ) {
```

Comment:
Is there a reason to filter this for admin users? Perhaps we should just short-circuit in case of admins. Thoughts?

Reply:
Yeah we could short-circuit if `current_user_can( 'unfiltered_html' )`

Reply:
Yep, this way we'd include Super Admins, too. Will add.